src/main/java/com/uber/cadence/internal/replay/ClockDecisionContext.java

```diff
@@ -88,20 +86,23 @@ public void accept(Exception reason) {
   private final Map<String, OpenRequestInfo<byte[], ActivityType>> pendingLaTasks = new HashMap<>();
   private final Map<String, ExecuteLocalActivityParameters> unstartedLaTasks = new HashMap<>();
   private final ReplayDecider replayDecider;
+  private final DataConverter dataConverter;
   private final Lock laTaskLock = new ReentrantLock();
   private final Condition taskCondition = laTaskLock.newCondition();
   private boolean taskCompleted = false;
 
   ClockDecisionContext(
       DecisionsHelper decisions,
       BiFunction<LocalActivityWorker.Task, Duration, Boolean> laTaskPoller,
-      ReplayDecider replayDecider) {
+      ReplayDecider replayDecider,
+      DataConverter dataConverter) {
```

Comment:
I believe DataConverter should live in higher level. At this level everything is a byte. Look at getVersion call in this file.

Reply:
The difference is getVersion is from workflow code, so we can access data converter from SyncDecisionContext and pass it down.
Here deserialization needs to happen while we replay event. So we will need to wire data converter into replay decider.